code/game/objects/items/weapons/power_cells.dm

```diff
@@ -247,6 +248,28 @@
 		for(var/mob/living/L in view(get_turf(src), max(5,(maxcharge/charge))))
 			L.apply_radiation(charge_rate/10, RAD_EXTERNAL)
 
+/obj/item/weapon/cell/rad/emp_act(severity)
+	..()
+	switch(rand(3))
+		if(0)
+			charge_rate *= severity*0.3
+			damaged = TRUE
+		if(1)
+			maxcharge *= severity*0.3
+			charge = 0
+		if(2)
+			maxcharge *= severity*0.3
+			charge = 0
+			charge_rate *= severity*0.3
+			damaged = TRUE
+		if(3)
+			return
+
+/obj/item/weapon/cell/rad/examine(mob/user)
+	..()
+	if(damaged)
```

Comment:
Rather than adding another bool, you can just check if charge_rate is less than initial charge rate

Reply:
Is that a better solution? I just stitched the thing together with exxion's help.

Reply:
It's possible to roll damage that doesn't reduce the charge rate, so IMO it's good to indicate that even if the message isn't perfect